### DISCUSSION TITLE
When using the defaultValues array and controller, what is an empty in dirtyFiles?

### DISCUSSION THREAD

**Discussion by dev-splin** [react-hook-form/react-hook-form#9898] [Q&A] [answered]
https://codesandbox.io/s/dirtyfields-undefined-rm1wkc?file=/src/index.js

Logs are too different when changing the 3rd selection box in the code above and when changing the 6th selection box.

If there is only one changed value, shouldn't there be only one dirtyFields?

Why does undefined appear randomly in dirtyFields?

**Comment by leapful** [comment, marked as answer]
@dev-splin 

No, `undefined` value does not appear randomly in  `dirtyFields` object. 

You are using `useFieldArray` hook to handle list item of field name `test.${index}.ReactSelect`, therefore, `test` should be an Array , a.k.a , `[]`. While dirty fields represents if the correspondent field name has dirty value or not with boolean true / false, it also represents the structure of field too.

For example:

```
const formValues = {
   firstName: 'A',
   lastName: 'B',
   profile: {
      email: 'email@email.com'
   },
   options: [
       { item: 'C' },
       { item: 'D' },
   ]
}

// Dirty fields should be
{
   firstName: true,
   lastName: true,
   profile: {
      email: true
   },
   options: [
      { item: true },
      { item: true },
   ]
}
```

In your case, there are only selection box 3 and 6 changed, so only them exists in `test` array but then the dirty field for array `test` should ensure the order too, otherwise, how can you determine what item is changed in the array ?

**Reply by dev-splin** [reply]
I read the post late. Thank you. I understand!
